Add a preference to open links in SFSafariViewController or Chrome

bval opened the pull request:
In iOS 9 Apple introduced the SFSafariViewController and it's since spread to become the default way to open a link from inside your app. This PR implements that for Colloquy mobile, makes it the default, and allows users to switch back to the old behavior of calling openURL instead.

The PR also goes a step further allowing users to open their links in Chrome if installed, and if they choose to do that and Chrome is not installed, it will send them to the App Store to install Chrome.

I am not much of an iOS developer but I _think_ I did a decent job here. Hope to get some constructive feedback though.

![simulator screen shot - iphone x - 2017-11-25 at 14 13 28](https://user-images.githubusercontent.com/57629/33234394-c62853da-d1eb-11e7-9249-e5aaefeb6cf2.png)
![simulator screen shot - iphone x - 2017-11-25 at 14 13 45](https://user-images.githubusercontent.com/57629/33234395-c84be4c4-d1eb-11e7-928f-0d0dd6490bd5.png)
![simulator screen shot - iphone x - 2017-11-25 at 14 17 53](https://user-images.githubusercontent.com/57629/33234396-ca3107d8-d1eb-11e7-9388-6a9958f9e536.png)

zadr commented:
Hi, this looks great, thanks for sending it in! I do have two questions about things in here;

1. If the preference is set to use Chrome, this looks like it will try to open up all links in Google Chrome, and then only continue if we have an http or https link. What happens for non-http links, such as spotify: URI’s, or mail: links (or even a link with caps HTTP://)? I haven’t had a chance to pull these changes into a branch and run them yet, but reading through the code, it looks like all these links will silently fail to open, which would be bad!

2. When you modified Root.plist to add this new item, the ability to navigate to Advanced Behavior was removed. I’m guessing this was an accident! Could you add it back in please?

Thanks again for sending this in, and please let me know (in here, or on irc, irc://chat.freenode.com/colloquy-mobile) if you have any questions.

bval commented:
@zadr Thanks for taking a look! You're right on both counts.

Colloquy won't be able to open any schemes not listed in `LSApplicationQueriesSchemes` so I thought sticking to http/https would be safe, but didn't think about `mailto:` or `tel:`. To accommodate (1) I've added an else case to the Chrome branch of that if, and switched to using `caseInsensitiveCompare`.

(2) was totally an oversight and I've added it back in.

zadr commented:
Re: `LSApplicationQueriesSchemes`: Apps can still ask iOS to open any URL they want, the limitation is around _checking_ to see if an app can open a URL or not before opening it.

That is, apps can always say "Hey, iOS, open up `spotify:artist:5lsC3H1vh9YSRQckyGv0Up`" and then if Spotify is installed, the link will be open.

Since we don't explicitly list `spotify` in `LSApplicationQueriesSchemes`, you are right that we couldn't say "Hey iOS, can we open a Spotify URL? If so, open this link up. Otherwise, we'll open  Music."

zadr commented:
Anyway, thanks for tweaking the PR, I've gone and merged this in 😊

bval commented:
🎉 Officially my first contribution in ObjC!

Thanks for your feedback on the PR. Feels Goodman.
